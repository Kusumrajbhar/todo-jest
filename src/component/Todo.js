import React from "react";

const Todo = ({ todo }) => {
  const { id, dish, completed } = todo;
  const h1 = <h1>{dish}</h1>;
  const paid = completed ? <strike>{dish}</strike> : h1;

  return (
    <div data-testId={`todo-${id}`}>
      <h1>{paid}</h1>
    </div>
  );
};

export default Todo;
