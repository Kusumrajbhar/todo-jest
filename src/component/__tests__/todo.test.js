import { cleanup, render, screen } from "@testing-library/react";
import Todo from "../Todo";
import renderer from "react-test-renderer";

afterEach(() => {
  cleanup();
});

test("should render not completed Todo", () => {
  const todo = {
    id: 1,
    dish: "Pizza",
    completed: false,
  };
  render(<Todo todo={todo} />);
  const todoElement = screen.getByTestId("todo-1");
  expect(todoElement).toBeInTheDocument();
  expect(todoElement).toHaveTextContent("Pizza");
  expect(todoElement).not.toContainHTML("<strike>");
});

test("should render completed Todo", () => {
  const todo = {
    id: 2,
    dish: "Burger",
    completed: true,
  };
  render(<Todo todo={todo} />);
  const todoElement = screen.getByTestId("todo-2");
  expect(todoElement).toBeInTheDocument();
  expect(todoElement).toHaveTextContent("Burger");
  expect(todoElement).not.toContainHTML("<strike>");
});

test("matches snapshot", () => {
  const todo = { id: 1, dish: "Pizza", completed: false };
  const tree = renderer.create(<Todo todo={todo} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("matches second snapshot", () => {
  const todo = { id: 2, dish: "Burger", completed: true };
  const tree = renderer.create(<Todo todo={todo} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("matches third snapshot", () => {
  const todo = { id: 3, dish: "Sandwich", completed: true };
  const tree = renderer.create(<Todo todo={todo} />).toJSON();
  expect(tree).toMatchSnapshot();
});
