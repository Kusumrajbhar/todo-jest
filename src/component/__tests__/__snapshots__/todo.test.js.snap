// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testId="todo-1"
>
  <h1>
    <h1>
      Pizza
    </h1>
  </h1>
</div>
`;
